### ISSUE TITLE
SOLUTION: Temporary Fix for Version 3.0.5 (login files not updating)

### ISSUE THREAD

**Issue by ubereffect** [TheCheatsrichter/Gw2_Launchbuddy#375]
[Updated] - [Launchbuddy Version 3.0.6](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/releases/tag/3.0.6) has since been released. I'm hopeful that the update will put this issue to rest. Be aware that your security software may not like the new Gw2.Launchbuddy.exe file, see [Megathread - Antivirus False Positives #363](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/issues/363).

-----

[2nd Update] - [Launchbuddy Version 3.0.6](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/releases/tag/3.0.6) successfully updated my login/dat files following two recent GW2 game updates. I'm no longer using 2 versions of Launchbuddy. Looking forward to 3.1.0, which is to include many bug fixes.

-----

PROBLEM: Since Version 3.0.5 Launchbuddy doesn't auto-update login / Gw2.dat files. The Launchbuddy application window may still load up (at least some people seem to have reported such), or the application may end up crashing before you even get to see the application window (which has always been my experience). Either way, the Gw2.dat files won't update, and you can't launch game instances until your dat files are current.

SOLUTION: Slightly different scenarios / solutions bellow.

Launchbuddy application window loads up without crashing: Click 'Account Settings' on the left hand margin. Select an account, then click on 'Login Credentials'. Click the 'Repair Loginfile' button. Repeat this process for each of your accounts. No need to re-renter login info, it only takes a few clicks and a short wait per account.

Launchbuddy crashes before the application window loads up: Download [Launchbuddy Version 2.4.2](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/releases/tag/2.4.2) and use it to update your login / Gw2.dat files. It may auto-update all of your login / Gw2.dat files without issue. Otherwise, if you get error messages, work your way through them (for all of your accounts), and the Launchbuddy application window should load up. At this point you can follow the same steps (above) to repair your login /Gw2.dat files.

If none of the above works: Launchbuddy saves information under Users\Your User Name\AppData\Roaming\Gw2 Launchbuddy. You can get half way there, by using search or run in Windows, typing %Appdata% and hitting Enter or Ok. You'll want to delete your Accs.xml file within this folder, and delete your dat files in the Loginfiles subfolder. Back these files up if you like. Once deleted, you should be able to recreated your login / Gw2 dat files using 3.0.5. If you take this approach, you'll have to re-enter all of your login information.

NOTE: Don't be fooled when you see 'Loginfile successfully configured' under Account Settings / which ever account you selected / Login Credentials. I've had it display this message when my login / Gw2.dat files still needed to be repaired / updated.

**Comment by ubereffect** [comment]
ADDITIONAL INFO

You can keep multiple versions of Launchbuddy on your PC. I currently have 3.0.5 and 2.4.2 in separate folders, with shortcuts for each. That way I can easily run 2.4.2 whenever my login / Gw2.dat files need updating, and revert back to 3.0.5 once that's done. Simply renaming the Gw2.Launchbuddy.exe also works, that way you can keep them both in the same folder, or on your desktop if you prefer.

If you rather just use a single version of Launchbuddy, you may have some luck sticking with an older version. I've been using 2.4.2 to auto-update my login / Gw2.dat file without issue for a few game updates, and imagine I could have just used that. But with the most recent game update, I got a bunch of errors and had to manually repair my login / Gw2.dat files. At least it didn't crash. [TheWhiteWolves](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/issues/370#issuecomment-1232010996) reported moments ago that 3.0.4 is working fine for them. I found that version to be painfully slow at launching multiple accounts however. Launchbuddy's version history can be found [here](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/releases).

Thanks again to [akeck](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/issues/368#issuecomment-1229076706) for first pointing out the possibility of using v2 to update your files, when v3 fails to do so.

**Comment by ubereffect** [comment]
Hoping this 'New issue' / thread will make it easier for people to quickly find the solution they're looking for.

**Comment by accelleratii** [comment]
1. 3.0.5 repair, breaks more than 50% of the files 100% of the time (small sample size).
2. 2.4.2 and 3.0.4 are reportedly working - still testing the solution.
3. Deleting .dat files does not reset errors or remove actual entries in the LB app. You will have to go for the settings file or the whole folder, which makes little difference anyway.

**Comment by TheUrps** [comment]
Unfortunately non of the solutions worked for me. 

I'm gonna try installing version 2.4.2.

**Comment by ubereffect** [comment]
> Unfortunately non of the solutions worked for me.
> 
> I'm gonna try installing version 2.4.2.
> 
> Edit: Nope, not even 2.4.2 is working anymore. I can't create Login-Files as the "Please check remember email/password and press the login and play button. [etc.]" windows stay present indefinitely.

In hindsight, I should of noted the error messages I got while running 2.4.2 following the most recent update. Don't recall what they said, or what I clicked to make them go away. I do remember having to do so over and over again, as Launchbuddy worked it's way through my accounts. I also believe I closed the Gw2 game launcher each time. Once it finally attempted all of my accounts, the Launchbuddy application window popped up, and I was able to repair my files.

Results may vary. Went from auto-updating, to errors -> repair, for me at least. If unable to update through 2.4.2, maybe worth trying another version like 3.0.4 perhaps.

**Comment by TheCheatsrichter** [comment]
Pushed 3.0.6 with updated File Update, Login and Gamestart timings.
Please give it a try and report back if it works now.

**Comment by ubereffect** [comment]
> Pushed 3.0.6 with updated File Update, Login and Gamestart timings. Please give it a try and report back if it works now.

So far so good. Security software isn't happy, as expected, see #363. Other than that, no issues here.
I suppose the next Game Update will be the real test. 

Thanks for the update.

**Comment by ubereffect** [comment]
Updated my original post to reflect the availability of [3.0.6](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/releases/tag/3.0.6)

**Comment by TheUrps** [comment]
Oddly enough now there's a new bug. When I press "Set Loginfile" the small Launchbuddy Windows opens, the GW2-Launcher pops up for a split seconds then it disappears and says "Loginfile successfuly configured", except it didn't.

**Comment by TheCheatsrichter** [comment]
Several code overhauls and changes have been made since 3.0.6.
Currently locked behind patreon exclusivity. But they will be pushed with version 3.1 as soon at it is finished.

eafc4e350e99390e56c42be9158abf193d366c72
2fad5247d19fcfd194437a7db321f75d65e08872
3d26c9f6c620bbdbe9a25f94d5212f5e3108378f
39a2c50b37c5df96a45940bbe85c4b4e60ce2efc
0eb8e3ed208b722243444ffe4ce9717d76a9cede

**Comment by ubereffect** [comment]
[Launchbuddy Version 3.0.6](https://github.com/TheCheatsrichter/Gw2_Launchbuddy/releases/tag/3.0.6) successfully updated my login/dat files following two recent GW2 game updates. I'm no longer using 2 versions of Launchbuddy. Looking forward to 3.1.0, which is to include many bug fixes. (Updated my original post)
